pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/flags.go

```diff
@@ -174,37 +175,42 @@ func ConfigToFlags(kubeAPIServerConfig *kubecontrolplanev1.KubeAPIServerConfig)
 	return arguments, nil
 }
 
-// currently for cluster up, audit is just broken.
-// TODO fix this
 func auditFlags(kubeAPIServerConfig *kubecontrolplanev1.KubeAPIServerConfig) map[string][]string {
-	setAudit := false
 	args := map[string][]string{}
+
 	for key, slice := range kubeAPIServerConfig.APIServerArguments {
 		if !strings.HasPrefix("audit-", key) {
 			continue
 		}
-		setAudit = true
 		for _, val := range slice {
 			args[key] = append(args[key], val)
 		}
 	}
-	if setAudit {
-		return args
-	}
 
-	// if the user has not attempted to set any audit values and the user has not explicitly said to disable audit logging
-	// then we set a default audit to stdout
-	defaultAuditPolicyFile := "openshift.local.log/audit-policy.yaml"
-	if err := os.MkdirAll(filepath.Dir(defaultAuditPolicyFile), 0755); err != nil {
-		utilruntime.HandleError(err)
-	}
-	if err := ioutil.WriteFile(defaultAuditPolicyFile, []byte(defaultAuditPolicy), 0644); err != nil {
-		utilruntime.HandleError(err)
-	} else {
-		setIfUnset(args, "audit-log-maxbackup", "10")
-		setIfUnset(args, "audit-log-maxsize", "100")
-		setIfUnset(args, "audit-log-path", "openshift.local.log/audit.log")
-		setIfUnset(args, "audit-policy-file", defaultAuditPolicyFile)
+	if kubeAPIServerConfig.AuditConfig.Enabled {
+		auditPolicyFilePath := kubeAPIServerConfig.AuditConfig.AuditFilePath
+		if len(kubeAPIServerConfig.AuditConfig.PolicyConfiguration.Raw) > 0 && string(kubeAPIServerConfig.AuditConfig.PolicyConfiguration.Raw) != "null" {
+			if len(kubeAPIServerConfig.AuditConfig.AuditFilePath) == 0 {
+				auditPolicyFilePath = "openshift.local.log/audit-policy.yaml"
+			}
+			if err := os.MkdirAll(filepath.Dir(auditPolicyFilePath), 0755); err != nil {
+				utilruntime.HandleError(err)
+			}
+			if err := ioutil.WriteFile(auditPolicyFilePath, kubeAPIServerConfig.AuditConfig.PolicyConfiguration.Raw, 0644); err != nil {
+				utilruntime.HandleError(err)
+			}
+		}
+
+		setIfUnset(args, "audit-log-maxbackup", strconv.Itoa(int(kubeAPIServerConfig.AuditConfig.MaximumRetainedFiles)))
+		setIfUnset(args, "audit-log-maxsize", strconv.Itoa(int(kubeAPIServerConfig.AuditConfig.MaximumFileSizeMegabytes)))
+		setIfUnset(args, "audit-log-maxage", strconv.Itoa(int(kubeAPIServerConfig.AuditConfig.MaximumFileRetentionDays)))
+		setIfUnset(args, "audit-log-path", kubeAPIServerConfig.AuditConfig.AuditFilePath)
```

Comment:
@soltysh what happens when this is not set?

Reply:
the log backend is disabled